### PR TITLE
Working Nav bar and Compose Feature

### PR DESCRIPTION
**Pull request by August0Rod**
Got compose feature working users are now able to use camera to take picture as well as the nav bar

### CODE REVIEW

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -6,6 +6,14 @@
     <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />
     <uses-permission android:name="android.permission.INTERNET" />
 
+    <queries>
+        <!-- Camera -->
+        <intent>
+            <action android:name="android.media.action.IMAGE_CAPTURE" />
```
Comment (ruiyangchen23) [inline app/src/main/AndroidManifest.xml:12]: just wonder, (i know this is the way to do it in code path), do you know the difference between intent for "android.media.action.IMAGE_CAPTURE" and "android.permission.CAMERA"? do you need to request permission whenever you want to use the camera or only the first time you use the app?

File: app/src/main/java/com/example/gymrats/fragments/ComposeFragment.java [hunk 2]
```diff
@@ -45,7 +46,7 @@ public class ComposeFragment extends Fragment {
     public static final String TAG = "ComposeFragment";
 
     //required variables for the compose page
-    public static final int CAPTURE_IMAGE_ACTIVITY_REQUEST_CODE = 42;
+    public static final int CAPTURE_IMAGE_ACTIVITY_REQUEST_CODE = 32;
```
Comment (ruiyangchen23) [inline app/src/main/java/com/example/gymrats/fragments/ComposeFragment.java:49]: is this used or will this be used?

**Review by ruiyangchen23 — APPROVED** [review]

File: app/src/main/java/com/example/gymrats/MainActivity.java [hunk 3]
```diff
@@ -28,22 +32,35 @@ public boolean onNavigationItemSelected(@NonNull MenuItem menuItem) {
...
+
                         break;
                     case R.id.action_exercise:
                         Toast.makeText(MainActivity.this, "Exercise!!", Toast.LENGTH_SHORT).show();
+                        fragment = new ComposeFragment();
+
                         break;
                     case R.id.action_profile:
                         Toast.makeText(MainActivity.this, "Profile!!", Toast.LENGTH_SHORT).show();
+                        fragment = new ComposeFragment();
```
Comment (ruiyangchen23) [inline app/src/main/java/com/example/gymrats/MainActivity.java:55]: can we merge the fragment = new ComposeFragment statement?

File: app/src/main/java/com/example/gymrats/Post.java [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+import com.parse.ParseClassName;
+import com.parse.ParseFile;
+import com.parse.ParseObject;
+import com.parse.ParseUser;
+
+import java.util.Date;
+
+@ParseClassName("Post")
+public class Post extends ParseObject {
```
Comment (ruiyangchen23) [inline app/src/main/java/com/example/gymrats/Post.java:14]: do we assume there is only one description/image/user in the database?

File: app/src/main/java/com/example/gymrats/fragments/ComposeFragment.java [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+    //required variables for the compose page
+    public static final int CAPTURE_IMAGE_ACTIVITY_REQUEST_CODE = 42;
+    private EditText etDescription;
+    private Button btnCaptureImage;
+    private ImageView ivPostImage;
+    private Button btnSubmit;
+
+    //gets photofile and photo file name for the fragment xml
+    private File photoFile;
+    public String photoFileName = "photo.jpg";
```
Comment (ruiyangchen23) [inline app/src/main/java/com/example/gymrats/fragments/ComposeFragment.java:57]: make it final?

**Review by ruiyangchen23 — APPROVED** [review]

File: app/src/main/java/com/example/gymrats/Post.java (hunk 4 above)
Comment (ruiyangchen23) [reply app/src/main/java/com/example/gymrats/Post.java:14]: i see, this is for post class. does the database use a post java object as some sort of key?

File: app/src/main/java/com/example/gymrats/fragments/ComposeFragment.java (hunk 5 above)
Comment (August0Rod) [reply app/src/main/java/com/example/gymrats/fragments/ComposeFragment.java:57]: Will do

File: app/src/main/java/com/example/gymrats/Post.java (hunk 4 above)
Comment (August0Rod) [reply app/src/main/java/com/example/gymrats/Post.java:14]: yeah i used the post class to get the key for the required from the database.

File: app/src/main/java/com/example/gymrats/fragments/ComposeFragment.java (hunk 2 above)
Comment (August0Rod) [reply app/src/main/java/com/example/gymrats/fragments/ComposeFragment.java:49]: i left the second one in there by accident, but the image call requires a arbitrary request code to be able to use the camera app inside the phone

File: app/src/main/java/com/example/gymrats/MainActivity.java (hunk 3 above)
Comment (August0Rod) [reply app/src/main/java/com/example/gymrats/MainActivity.java:55]: the new compose fragment is under different actions actions as a placeholder for now, once i have the other parts of the navigation bar set up, those will change to their specified fragments.

File: app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (August0Rod) [reply app/src/main/AndroidManifest.xml:12]: > 

i know that the  "android.media.action.IMAGE_CAPTURE" is used just to be able to capture images using the camera app, the permission Camera gives your app access to the camera feature with more functionality i believe. The way it was explained to us if you are using the built in camera app you do not need the "android.permission.CAMERA" unless you wish to make your own, which i plan to do in the future

File: app/src/main/java/com/example/gymrats/fragments/ComposeFragment.java (hunk 5 above)
Comment (August0Rod) [reply app/src/main/java/com/example/gymrats/fragments/ComposeFragment.java:57]: > make it final?
Just added this

File: app/src/main/java/com/example/gymrats/fragments/ComposeFragment.java (hunk 2 above)
Comment (August0Rod) [reply app/src/main/java/com/example/gymrats/fragments/ComposeFragment.java:49]: > is this used or will this be used?

just got rid of the extra variable

**Pull request merged**